x-pack/legacy/plugins/infra/public/containers/metrics/with_metrics_time.tsx

```diff
@@ -141,8 +150,21 @@ const mapToUrlState = (value: any): MetricsTimeUrlState | undefined =>
       }
     : undefined;
 
-const mapToTimeUrlState = (value: any) =>
-  value && (typeof value.to === 'number' && typeof value.from === 'number') ? value : undefined;
+const mapToTimeUrlState = (value: any) => {
```

Comment:
The value is actually unknown because it's coming in from the URl. I just updated the code for this function to be more robust and use `io-ts` to ensure the value is the correct shape and what we are expecting.

Reply:
Nice